tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,554 @@
+  });
+
+  it('always renders a disabled "email" input field', () => {
+    const root = renderUserProfileEdit();
+
+    expect(root.find('.UserProfileEdit-email')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-email')).toHaveProp('disabled', true);
+  });
+
+  it('renders help sections for some fields', () => {
+    const root = renderUserProfileEdit();
+
+    // Enzyme does not extract "text" from `dangerouslySetInnerHTML` prop,
+    // which is needed to escape the link in this help section.That is why we
+    // need the `toHaveHTML()` matcher below.
+    expect(root.find('.UserProfileEdit-email--help')).toHaveHTML(
+      oneLine`<p class="UserProfileEdit-email--help">You can change your email
+      address on Firefox Accounts.
+      <a href="https://support.mozilla.org/kb/change-primary-email-address-firefox-accounts">Need help?</a></p>`
+    );
+
+    expect(root.find('.UserProfileEdit-homepage--help')).toHaveText(
+      `This URL will only be visible for users who are developers.`
+    );
+
+    expect(root.find('.UserProfileEdit-biography--help')).toHaveText(
+      oneLine`Some HTML supported: <abbr title> <acronym title> <b>
+      <blockquote> <code> <em> <i> <li> <ol> <strong> <ul>. Links are
+      forbidden.`
+    );
+  });
+
+  it('renders a displayName input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-displayName')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-displayName'))
+      .toHaveProp('value', defaultUserProps.display_name);
+  });
+
+  it('renders a homepage input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-homepage')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-homepage'))
+      .toHaveProp('value', defaultUserProps.homepage);
+  });
+
+  it('renders a location input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-location')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-location'))
+      .toHaveProp('value', defaultUserProps.location);
+  });
+
+  it('renders a occupation input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-occupation')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-occupation'))
+      .toHaveProp('value', defaultUserProps.occupation);
+  });
+
+  it('renders a biography input field', () => {
+    const root = renderUserProfileEdit();
+    expect(root.find('.UserProfileEdit-biography')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-biography'))
+      .toHaveProp('value', defaultUserProps.biography);
+  });
+
+  it('captures input field changes ', () => {
+    const fields = [
+      'biography',
+      'displayName',
+      'homepage',
+      'location',
+      'occupation',
+      'username',
+    ];
+
+    const root = renderUserProfileEdit();
+
+    expect.assertions(fields.length);
```

Comment:
Yes! Because there is a loop below, I let Jest know that I expect `N` assertions, otherwise if something breaks (without an error, e.g., `break`) we will not notice anything.

Reply:
Nice!